methods.py

```diff
@@ -663,8 +664,13 @@ def find_visual_c_batch_file(env):
 
     # Scons 4.6.0+ removed passing env, so we need to get the product_dir ourselves first,
     # then pass that as the last param instead of env as the first param as before.
-    # We should investigate if we can avoid relying on SCons internals here.
-    product_dir = find_vc_pdir(env, msvc_version)
+    # Scons 4.8.0+ removed the env parameter from find_vc_pdir().
```

Comment:
4.8.0 can still take the env, it's just become the second and optional parameter:
```python
def find_vc_pdir(msvc_version, env=None):
```

So you could also just do `find_vc_pdir(msvc_version=msvc_version, env=env)`, but this would also break if they ever do remove `env` or decide to change parameter names next.

Reply:
```
    Args:
        msvc_version: str
            msvc version (major.minor, e.g. 10.0)
        env:
            optional to look up VSWHERE variable
```

I don't know if anyone uses it / how well we support the `VSWHERE` stuff, but if we do passing the optional env might still be desired.